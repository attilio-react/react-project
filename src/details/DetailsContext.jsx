import React from 'react';

export const DetailsContext = React.createContext(
   {
     movie: {},
     relatedMovies: []
   }
)


