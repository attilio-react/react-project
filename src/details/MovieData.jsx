import React from 'react';

import {DetailsContext} from './DetailsContext.jsx'

class MovieData extends React.PureComponent {
  render () {
    return (<DetailsContext.Consumer>
            {ctx => (
                <div>
                    <img src={ctx.movie.poster_path} />
                    <p><span>{ctx.movie.title}</span></p>
                    <p><span>{ctx.movie.vote_average}</span></p>
                    <p><span>{ctx.movie.release_date}</span></p>
                    <p><span>{ctx.movie.runtime}</span></p>
                    <p><span>{ctx.movie.overview}</span></p>
                </div>
            )}
            </DetailsContext.Consumer>)
  }

}


export {MovieData}


