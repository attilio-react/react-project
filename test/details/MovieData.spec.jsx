import * as React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';


import {MovieData} from '|details/MovieData.jsx';
import {DetailsContext} from '|details/DetailsContext.jsx';

describe('MovieData rendering', () => {
  it('Should render', () => {
    const 
        component = renderer.create(
            <DetailsContext.Provider value = {{
                movie: {
                    poster_path: 'http://hello.com/poster.png',
                    title: 'Movie Title',
                    vote_average: 4.4,
                    release_date: '2011-01-01',
                    runtime: 129,
                    overview: 'This is a movie about ....'
                }
            }}>
                <MovieData />
            </DetailsContext.Provider>
        ),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});


