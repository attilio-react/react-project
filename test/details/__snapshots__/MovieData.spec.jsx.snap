// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieData rendering Should render 1`] = `
<div>
  <img
    src="http://hello.com/poster.png"
  />
  <p>
    <span>
      Movie Title
    </span>
  </p>
  <p>
    <span>
      4.4
    </span>
  </p>
  <p>
    <span>
      2011-01-01
    </span>
  </p>
  <p>
    <span>
      129
    </span>
  </p>
  <p>
    <span>
      This is a movie about ....
    </span>
  </p>
</div>
`;
